Handy macros for JACoW editors

zhichu opened the pull request:
# ISO-4 formatter
JACoW prefers to use the ISO-4 abbreviation for the journal title. This macro converts the selected text to the ISO-4 form.

## Prerequisite
`cURL` 

# Format `thebibliography`
Citation order should be in sequence. This macro automatically sort `thebibliography` environment, creating dummy ones for missing entries and removing the ones not cited in text. The original environment will be put at the end of the document for comparison.

# SI formatter
Convert selected text to `\SI{num}{unit}` or `\si{unit}`.

zhichu committed:
Handy macros for JACoW events

zhichu committed:
Update SI formatter.txsMacro

Add “°” case

zhichu commented:
Dear TeXstudio developers,

I'm from the JACoW Team and TeXstudio is the de facto editor for us editors. We are under pressure to format hundreds if not thousands of papers submitted by authors who do not always follow our style, during a typically 4-5 days event. I hope we can use some macros to make our work easier. It will be great if we don't have to manually import them at each event.

Love the project, by the way.
Zhichu

sunderme commented on jacow/ISO-4 formatter.txsMacro:
```diff
@@ -0,0 +1,24 @@
+{
+    "abbrev": "",
+    "description": [
+        "Format the selected text in ISO-4 form."
```
This description should be a bit more extended, especially since it calls an external tool/webpage

sunderme replied:
txs users will only see this description ...

sunderme commented on jacow/SI formatter.txsMacro:
```diff
@@ -0,0 +1,34 @@
+{
+    "abbrev": "",
+    "description": [
+        "Format the selected text with `siunitx`."
```
maybe a bit extended 
an example may be helpful

zhichu committed:
Update ISO-4 formatter.txsMacro

Add description to the macro

zhichu committed:
Update SI formatter.txsMacro

Add description and fix regular expression and handle exception for none matching situation

zhichu reviewed (commented):
Fix a problem and improve documentation

sunderme commented:
Thanks
